src/mongocxx/collection.cpp

```diff
@@ -300,29 +300,25 @@ stdx::optional<result::delete_result> collection::delete_one(
     return stdx::optional<result::delete_result>(result::delete_result(std::move(result.value())));
 }
 
-stdx::optional<bsoncxx::document::value> collection::find_one_and_replace(
-    bsoncxx::document::view filter, bsoncxx::document::view replacement,
-    const options::find_one_and_replace& options) {
+stdx::optional<bsoncxx::document::value> _find_and_modify(::mongoc_collection_t* collection,
+                                                          bsoncxx::document::view filter,
+                                                          ::mongoc_find_and_modify_opts_t* opts) {
     scoped_bson_t bson_filter{filter};
-    scoped_bson_t bson_replacement{replacement};
-    scoped_bson_t bson_sort{options.sort()};
-    scoped_bson_t bson_projection{options.projection()};
-
     scoped_bson_t reply;
     reply.flag_init();
 
-    bson_error_t error;
+    ::bson_error_t error;
 
-    options::return_document rd =
-        options.return_document().value_or(options::return_document::k_before);
+    bool r = libmongoc::collection_find_and_modify_with_opts(collection, bson_filter.bson(), opts,
+                                                             reply.bson(), &error);
 
-    bool r = mongoc_collection_find_and_modify(
-        _impl->collection_t, bson_filter.bson(), bson_sort.bson(), bson_replacement.bson(),
-        bson_projection.bson(), false, options.upsert().value_or(false),
-        rd == options::return_document::k_after, reply.bson(), &error);
+    libmongoc::find_and_modify_opts_destroy(opts);
 
     if (!r) {
-        throw exception::write(std::move(_impl->gle()), std::make_tuple(error.message, error.code));
+        const ::bson_t* gle = libmongoc::collection_get_last_error(collection);
+        libbson::scoped_bson_t error_copy;
```

Comment:
I'd fully qualify the calls to scoped_bson_t, or not (I'd lean on the side of not, following the lead for the rest of the file)